Automattic/wp-calypso
# Adds the possibility to use the WPCOM theme in the Blaze Dashboad app

## Proposed Changes

This PR will handle a new parameter, `is_running_in_blaze_plugin`, that we will use to select the dashboard theme.
The new possible themes are: wpcom, woo and jetpack.

## Why are these changes being made?

As part of our work for the Blaze Ads plugin, we plan to release it to non-woo commerce sites. In those cases, we want to brand it like in WPCOM (theme, name, etc). 

## Testing Instructions

You will need a WPCOM Sandbox to test this PR
* Checkout this branch
* On a console, connect to your sandbox
* On another console, navigate to `wp-calypso/apps/blaze-dashboard` and execute `yarn dev --sync.`
* You will also need to move the traffic from `widgets.wp.com` to your sandbox (in your host file)
* Go to the Jetpack Blaze Dashboard for one of your sites. Any Jetpack-connected self-host site will work, and some atomics are already using the Jetpack Blaze version of the dashboard).
* Verify that you see the correct Jetpack theme:
![Screenshot 2024-07-29 at 2 08 34 PM](https://github.com/user-attachments/assets/9a219d57-4969-4bdf-be96-b6416671ec9b)
* Now, go to the file `apps/blaze-dashboard/src/load-config.js` and force the `isBlazePlugin` flag to be true.
* Refresh the page, and verify that  you see the WPCOM theme
![Screenshot 2024-07-29 at 2 10 37 PM](https://github.com/user-attachments/assets/8d419081-6d71-421d-8cd9-ee54eb42cde4)
* On the same file, force both the `isBlazePlugin` and `isWooStore` to true
* Refresh the page and verify that you see the WooCommerce version of the page
![Screenshot 2024-07-29 at 2 12 07 PM](https://github.com/user-attachments/assets/2c03cf49-f735-44f6-b3be-08440eee7b27)

Thats it! As an extra, if you already have a site running with the new Blaze Ads plugin, you can check that still shows the correct version of the page.

## Pre-merge Checklist
